OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -201,7 +198,7 @@ string IOrderVoice.VoicePhraseForOrder(Actor self, Order order)
 
 		public abstract Activity GetAttackActivity(Actor self, Target newTarget, bool allowMove, bool forceAttack);
 
-		public bool HasAnyValidWeapons(Target t)
+		public bool HasAnyValidWeapons(Target t, bool onlyAttackCenterWeapons)
```

Comment:
Better to define this as `bool onlyAttackCenterWeapons = false` so that you don't need to explicitly pass `false` in from all but one place.